packages/flutter_tools/lib/src/macos/native_assets.dart

```diff
@@ -157,9 +161,107 @@ Asset _targetLocationMacOS(Asset asset, Uri? absolutePath) {
+  Uri buildUri,
+  Map<AssetPath, List<Asset>> assetTargetLocations,
+  String? codesignIdentity,
+  BuildMode buildMode,
+  FileSystem fileSystem,
+) async {
+  if (assetTargetLocations.isNotEmpty) {
+    globals.logger.printTrace('Copying native assets to ${buildUri.toFilePath()}.');
+    for (final MapEntry<AssetPath, List<Asset>> assetMapping in assetTargetLocations.entries) {
+      final Uri target = (assetMapping.key as AssetAbsolutePath).uri;
+      final List<Uri> sources = <Uri>[for (final Asset source in assetMapping.value) (source.path as AssetAbsolutePath).uri];
+      final Uri targetUri = buildUri.resolveUri(target);
+      final String name = targetUri.pathSegments.last;
+      final Directory frameworkDir = fileSystem.file(targetUri).parent;
+      if (await frameworkDir.exists()) {
+        await frameworkDir.delete(recursive: true);
+      }
+      // MyFramework.framework/                           frameworkDir
+      //   MyFramework  -> Versions/Current/MyFramework   dylibLink
+      //   Resources    -> Versions/Current/Resources     resourcesLink
+      //   Versions/                                      versionsDir
+      //     A/                                           versionADir
+      //       MyFramework                                dylibFile
+      //       Resources/                                 resourcesDir
+      //         Info.plist
+      //     Current  -> A                                currentLink
+      final Directory versionsDir = frameworkDir.childDirectory('Versions');
+      final Directory versionADir = versionsDir.childDirectory('A');
+      final Directory resourcesDir = versionADir.childDirectory('Resources');
+      await resourcesDir.create(recursive: true);
+      final File dylibFile = versionADir.childFile(name);
+      final Link currentLink = versionsDir.childLink('Current');
+      await currentLink.create(fileSystem.path.relative(versionADir.path, from: currentLink.parent.path));
+      final Link resourcesLink = frameworkDir.childLink('Resources');
+      await resourcesLink.create(fileSystem.path.relative(resourcesDir.path, from: resourcesLink.parent.path));
+      await lipoDylibs(dylibFile, sources);
+      final Link dylibLink = frameworkDir.childLink(name);
+      await dylibLink.create(fileSystem.path.relative(versionsDir.childDirectory('Current').childFile(name).path, from: dylibLink.parent.path));
+      await setInstallNameDylib(dylibFile);
+      await createInfoPlist(name, resourcesDir);
+      await codesignDylib(codesignIdentity, buildMode, frameworkDir);
+    }
+    globals.logger.printTrace('Copying native assets done.');
+  }
+}
+
+
+/// Copies native assets for flutter tester.
+///
+/// For `flutter run -release` a multi-architecture solution is needed. So,
+/// `lipo` is used to combine all target architectures into a single file.
+///
+/// The install name is set so that it matches what the place it will
+/// be bundled in the final app.
```

Comment:
Is this comment true? Looks like you're not changing install name in this function?

Reply:
Also, can you add a comment on how and why this differs from `_copyNativeAssetsMacOS`?